src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,1001 @@
+    var counts = gpii.devpmt.calculateCommonTermUsageCounts(commonTermsSorted, flatPrefs);
+    that.applier.change("commonTermUsageCounts", counts);
+};
+
+/**
+ * Lookup Generic Preference Setting.
+ *
+ * Using the context and commonTerm lookup the generic preference
+ * value. If the preference set does not contain the value, we
+ * return `undefined` to make this known, as `undefined` is not
+ * a valid JSON value. Theoretically, and preference setting maybe
+ * set as `null`.
+ */
+gpii.devpmt.lookupGenericPrefValue = function (that, context, commonTerm) {
+    if (that.model.flatPrefs.contexts[context].preferences[commonTerm] !== undefined) {
+        return that.model.flatPrefs.contexts[context].preferences[commonTerm];
+    }
+    else {
+        return undefined;
+    }
+};
+
+/**
+ * Lookup Application Specific Setting. Similar to `lookupGenericPrefValue` lookups
+ * in regard to return values, but takes an extra argument for the product.
+ */
+gpii.devpmt.lookupProductPrefValue = function (that, context, product, settingTerm) {
+    if (that.model.flatPrefs.contexts[context].preferences[product] &&
+        that.model.flatPrefs.contexts[context].preferences[product][settingTerm] !== undefined) {
+        return that.model.flatPrefs.contexts[context].preferences[product][settingTerm];
+    }
+    else {
+        return undefined;
+    }
+};
+
+gpii.devpmt.lookupPspMetadata = function (that, type, context, term, product) {
+    if (product) {
+        if (that.model.flatPrefs.contexts[context].metadata &&
+            that.model.flatPrefs.contexts[context].metadata.psp &&
+            that.model.flatPrefs.contexts[context].metadata.psp[type] &&
+            that.model.flatPrefs.contexts[context].metadata.psp[type][product] &&
+            that.model.flatPrefs.contexts[context].metadata.psp[type][product][term]) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+    else if (that.model.flatPrefs.contexts[context].metadata &&
+        that.model.flatPrefs.contexts[context].metadata.psp &&
+        that.model.flatPrefs.contexts[context].metadata.psp[type] &&
+        that.model.flatPrefs.contexts[context].metadata.psp[type][term]) {
+        return true;
+    }
+    return false;
+};
+
+gpii.devpmt.initSettingTableWidgets = function (that) {
+    var existingTables = fluid.queryIoCSelector(fluid.rootComponent, "gpii.devpmt.settingsTableWidget");
+    fluid.each(existingTables, function (item) {
+        item.destroy();
+    });
+    fluid.each(that.model.npsetApplications, function (item) {
+        var sel = "#test-comp-" + item.appId;
+        sel = sel.replace(/\./g, "\\.");
+        var appUri = "http://registry.gpii.net/applications/" + item.appId;
+        that.events.createSettingsTable.fire(sel, item.appId, that.model.allSolutions[item.appId], appUri);
+    });
+};
+
+gpii.devpmt.addEditToUnsavedList = function (that, description) { // that, path, newValue, oldValue) {
```

Comment:
Remove the comment if it's no longer needed.

Reply:
done